keylime/config.py

```diff
@@ -242,9 +242,10 @@ def list_to_dict(alist):
     return params
 
 
-def yaml_to_dict(arry):
+def yaml_to_dict(arry, add_newlines=True):
     arry = convert(arry)
-    return yaml.load("\n".join(arry), Loader=SafeLoader)
+    sep = "\n" if add_newlines else ""
+    return yaml.load(sep.join(arry), Loader=SafeLoader)
```

Comment:
FYI, we can see here that `yaml_to_dict` does not necessarily return a `dict`; it can return anything that `yaml.load` can.

Reply:
Then maybe `yaml_to_dict` needs to verify what `yaml.load` is returning and fail if it's not a dict.

Reply:
I looked at all the uses of `yaml_to_dict`, and it looks to me like they do indeed assume that the result is a dict (or maybe, in some cases, a value whose truthiness is false).  So, yes, I think that adding a check inside `yaml_to_dict` makes sense; it could raise an Exception when the result is not a dict.